spec/models/claim/matching_attribute_finder_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe Claim::MatchingAttributeFinder do
+  describe "#claim_ids_with_matching_attributes" do
+    let(:source_claim) {
+      create(
+        :claim,
+        teacher_reference_number: "0902344",
+        national_insurance_number: "QQ891011C",
+        email_address: "genghis.khan@mongol-empire.com",
+        bank_account_number: "34682151",
+        bank_sort_code: "972654",
+        building_society_roll_number: "123456789/ABCD",
+      )
+    }
+
+    let!(:claim_with_no_matching_attributes) { create(:claim, :submitted) }
+    let!(:unsubmitted_claim_with_matching_teacher_reference_number) { create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number) }
+
+    let(:claim_with_matching_attribute) { build(:claim, :submitted) }
+    let(:claims_with_matching_attributes) { Claim::MatchingAttributeFinder.new(source_claim).claim_ids_with_matching_attribute_keys }
+
+    it "does not include the id of the source claim" do
+      expect(claims_with_matching_attributes.keys).not_to include(source_claim.id)
+    end
+
+    it "does not include the id of a claim without any matcheing attributes" do
+      expect(claims_with_matching_attributes.keys).not_to include(claim_with_no_matching_attributes.id)
+    end
+
+    it "only includes submitted claims" do
+      expect(claims_with_matching_attributes.keys).not_to include(unsubmitted_claim_with_matching_teacher_reference_number.id)
+    end
+
+    it "includes the id of a claim with a mathching teacher reference number" do
+      claim_with_matching_attribute.teacher_reference_number = source_claim.teacher_reference_number
+      claim_with_matching_attribute.save
+
+      expect(claims_with_matching_attributes.keys).to include(claim_with_matching_attribute.id)
+      expect(claims_with_matching_attributes[claim_with_matching_attribute.id]).to include("Teacher reference number")
+    end
+
+    it "includes the id of a claim with a mathching national insurance number" do
+      claim_with_matching_attribute.national_insurance_number = source_claim.national_insurance_number
+      claim_with_matching_attribute.save
+
+      expect(claims_with_matching_attributes.keys).to include(claim_with_matching_attribute.id)
+      expect(claims_with_matching_attributes[claim_with_matching_attribute.id]).to include("National insurance number")
```

Comment:
This is usually capitalised as "National Insurance number". I wonder whether it would be better if `MatchingAttributeFinder` didn't try to return human-readable attribute names, and instead returned the attribute names as the original symbols, e.g. `:national_insurance_number`. And then at the _view_ layer we could format them for presentation to the user.

Reply:
This probably isn't worth doing for now, especially given the fact that it's an internal screen